sycl/include/sycl/detail/spirv.hpp

```diff
@@ -906,15 +912,24 @@ EnableIfNativeShuffle<T> ShuffleXor(GroupT g, T x, id<1> mask) {
                                            static_cast<uint32_t>(mask.get(0)));
   }
 #else
+  T result;
   if constexpr (ext::oneapi::experimental::is_user_constructed_group_v<
                     GroupT>) {
-    return __nvvm_shfl_sync_bfly_i32(detail::ExtractMask(detail::GetMask(g))[0],
-                                     x, static_cast<uint32_t>(mask.get(0)),
-                                     0x1f);
+    auto MemberMask = detail::ExtractMask(detail::GetMask(g))[0];
+    if constexpr (is_fixed_size_group_v<GroupT>) {
+      CUDA_SHFL_SYNC(result, MemberMask, x, static_cast<uint32_t>(mask.get(0)),
+                     0x1f, bfly_i32)
+    } else {
+      int unfoldedSrcSetBit =
+          (g.get_local_id()[0] ^ static_cast<uint32_t>(mask.get(0))) + 1;
+      CUDA_SHFL_SYNC(result, MemberMask, x,
+                     __nvvm_fns(MemberMask, 0, unfoldedSrcSetBit), 31, idx_i32)
+    }
   } else {
-    return __nvvm_shfl_sync_bfly_i32(membermask(), x,
-                                     static_cast<uint32_t>(mask.get(0)), 0x1f);
+    CUDA_SHFL_SYNC(result, membermask(), x, static_cast<uint32_t>(mask.get(0)),
+                   0x1f, bfly_i32)
```

Comment:
Any reasons why in some places you use `31` and in others `0x1f`?

Reply:
Purely semantic/personal preference: 31 is used for shuffle down, to indicate the lane limit, 0x1f is used for bfly when the semantic is more meaningfully a "bit-mask".

Reply:
See https://docs.nvidia.com/cuda/parallel-thread-execution/#data-movement-and-conversion-instructions-shfl-sync for details